src/components/auth/UserAuthForm.tsx

```diff
@@ -0,0 +1,119 @@
+// src/components/auth/UserAuthForm.tsx
+import React, { useState } from 'react';
+import { useUserAuth } from '../../lib/UserAuthContext';
+
+export function UserAuthForm({ onSuccess }: { onSuccess?: () => void }) {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [isLogin, setIsLogin] = useState(true);
+  const [loading, setLoading] = useState(false);
+  const { login, register, loginWithGoogle, error } = useUserAuth();
+
+  const handleEmailAuth = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setLoading(true);
+
+    try {
+      if (isLogin) {
+        await login(email, password);
```

Comment:
Esto puede ser asi
```suggestion
        await login(userForm.email, userForm.password);
```
o pasarle directo el objeto de userForm